jupyterhub/handlers/pages.py

```diff
@@ -172,7 +172,19 @@ def _render_form(self, for_user, spawner_options_form, message=''):
                 spawner._spawn_future = None
             # not running, no form. Trigger spawn and redirect back to /user/:name
             f = asyncio.ensure_future(self.spawn_single_user(user, server_name))
-            await asyncio.wait([f], timeout=1)
+            done, pending = await asyncio.wait([f], timeout=1)
+            # If spawn_single_user throws an exception, raise a 500 error
+            # otherwise it may cause a redirect loop
+            if f in done:
```

Comment:
We only need to take this branch if f is done and failed:

```suggestion
            if f.done() and f.exception():
```